pycoast/cw_base.py

```diff
@@ -862,38 +864,47 @@ def add_overlay_from_dict(self, overlays, area_def, cache_epoch=None, background
 
         # Grids overlay
         if 'grid' in overlays:
-            lon_major = float(overlays['grid'].get('lon_major', 10.0))
-            lat_major = float(overlays['grid'].get('lat_major', 10.0))
-            lon_minor = float(overlays['grid'].get('lon_minor', 2.0))
-            lat_minor = float(overlays['grid'].get('lat_minor', 2.0))
-            font = overlays['grid'].get('font', None)
-            font_size = int(overlays['grid'].get('font_size', 10))
-            grid_kwargs = {}
-            if is_agg:
-                width = float(overlays['grid'].get('width', 1.0))
-                grid_kwargs["width"] = width
-
+            if 'major_lonlat' in overlays['grid'] or 'minor_lonlat' in overlays['grid']:
+                Dlonlat = overlays['grid'].get('major_lonlat', (10.0, 10.0))
+                dlonlat = overlays['grid'].get('minor_lonlat', (2.0, 2.0))
+            else:
+                Dlonlat = (overlays['grid'].get('Dlon', 10.0), overlays['grid'].get('Dlat', 10.0))
+                dlonlat = (overlays['grid'].get('dlon', 2.0), overlays['grid'].get('dlat', 2.0))
+            outline = overlays['grid'].get('outline', 'white')
             write_text = overlays['grid'].get('write_text', True)
             if isinstance(write_text, str):
                 write_text = write_text.lower() in ['true', 'yes', '1', 'on']
-            outline = overlays['grid'].get('outline', 'white')
+            # This should close PyCoast issue #52
+            font = overlays['grid'].get('font', None)
+            font_size = int(overlays['grid'].get('font_size', 10))
+            fill = overlays['grid'].get('fill', 'white')
+            fill_opacity = overlays['grid'].get('fill_opacity', 255)
             if isinstance(font, str):
                 if is_agg:
                     from aggdraw import Font
-                    font = Font(outline, font, size=font_size)
+                    font = Font(fill, font, opacity=fill_opacity, size=font_size)
```

Comment:
This is backwards incompatible, right? Anyone who was using `outline` color for the text color will now get a different/wrong color?

Reply:
Additionally, the `add_grid` method still uses `outline` from the command for the font that it creates in `_draw_grid_labels`.

Reply:
The problem is upt to now the lon/lat labels always had the color of the grid. You had to generate your font object in the satpy script If you wanted different colors. Backward compatibility is easy to get if 'fill' defaults to 'outline' (instead of 'white').

Reply:
Ah that's a nice idea. From what I can tell `fill` isn't used for anything else as far as grids go? If we were to release all of your fixes and feature ideas and then come out with a pycoast 2.0 that breaks backwards compatibility, do you think these keyword arguments should have different names?